$OpenBSD: patch-lib_rinda_ring_rb,v 1.1.1.1 2014/01/03 17:51:52 jeremy Exp $

Fix Rinda::RingFinger to work on OpenBSD, where IP_MULTICAST_LOOP
socket option takes unsigned char instead of int.

--- lib/rinda/ring.rb.orig	Thu Jan  2 20:09:44 2014
+++ lib/rinda/ring.rb	Thu Jan  2 20:09:52 2014
@@ -413,7 +413,7 @@ module Rinda
       soc = Socket.new(addrinfo.pfamily, addrinfo.socktype, addrinfo.protocol)
 
       if addrinfo.ipv4_multicast? then
-        soc.setsockopt(:IPPROTO_IP, :IP_MULTICAST_LOOP, true)
+        soc.setsockopt(Socket::Option.ipv4_multicast_loop(1))
         soc.setsockopt(:IPPROTO_IP, :IP_MULTICAST_TTL,
                        [@multicast_hops].pack('c'))
       elsif addrinfo.ipv6_multicast? then
